joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -42,6 +43,19 @@ trait AstCreatorHelper { this: AstCreator =>
   protected def prefixAsBuiltin(x: String): String = s"$builtinPrefix$pathSep$x"
   protected def pathSep                            = "."
 
+  protected def handleNewVariableOccurrence(node: RubyNode): Ast = {
+    val name = code(node)
+    scope.lookupVariable(name) match
+      case None =>
+        val local      = localNode(node, name, name, Defines.Any)
+        val identifier = identifierNode(node, name, name, Defines.Any)
+        scope.addToScope(name, local)
+        Ast(identifierNode(node, name, name, Defines.Any)).withRefEdge(identifier, local)
```

Comment:
Are you intentionally creating two separate identifierNodes here?

Reply:
Definitely not, good catch